connect/runtime/src/test/java/org/apache/kafka/connect/runtime/isolation/PluginUtilsTest.java

```diff
@@ -490,6 +504,168 @@ public void testPluginUrlsWithRelativeSymlinkForwards() throws Exception {
+                Collections.emptySortedSet(),
+                Collections.emptySortedSet(),
+                Collections.emptySortedSet(),
+                Collections.emptySortedSet(),
+                Collections.emptySortedSet()
+        );
+        Map<String, String> aliases = PluginUtils.computeAliases(result);
+        assertEquals(MockSinkConnector.class.getName(), aliases.get("MockSinkConnector"));
+        assertEquals(MockSinkConnector.class.getName(), aliases.get("MockSink"));
+    }
+
+    @Test
+    public void testCollidingPrunedAlias() {
+        SortedSet<PluginDesc<Converter>> converters = new TreeSet<>();
+        converters.add(new PluginDesc<>(CollidingConverter.class, null, CollidingConverter.class.getClassLoader()));
+        SortedSet<PluginDesc<HeaderConverter>> headerConverters = new TreeSet<>();
+        headerConverters.add(new PluginDesc<>(CollidingHeaderConverter.class, null, CollidingHeaderConverter.class.getClassLoader()));
+        PluginScanResult result = new PluginScanResult(
+                Collections.emptySortedSet(),
+                Collections.emptySortedSet(),
+                converters,
+                headerConverters,
+                Collections.emptySortedSet(),
+                Collections.emptySortedSet(),
+                Collections.emptySortedSet(),
+                Collections.emptySortedSet(),
+                Collections.emptySortedSet()
+        );
+        Map<String, String> aliases = PluginUtils.computeAliases(result);
+        assertEquals(CollidingConverter.class.getName(), aliases.get("CollidingConverter"));
+        assertEquals(CollidingHeaderConverter.class.getName(), aliases.get("CollidingHeaderConverter"));
+        assertFalse(aliases.containsKey("Colliding"));
+    }
+
+    @SuppressWarnings("unchecked")
+    @Test
+    public void testCollidingSimpleAlias() {
+        SortedSet<PluginDesc<Converter>> converters = new TreeSet<>();
+        converters.add(new PluginDesc<>(CollidingConverter.class, null, CollidingConverter.class.getClassLoader()));
+        SortedSet<PluginDesc<Transformation<?>>> transformations = new TreeSet<>();
+        transformations.add(new PluginDesc<>((Class<? extends Transformation<?>>) (Class<?>) Colliding.class, null, Colliding.class.getClassLoader()));
+        PluginScanResult result = new PluginScanResult(
+                Collections.emptySortedSet(),
+                Collections.emptySortedSet(),
+                converters,
+                Collections.emptySortedSet(),
+                transformations,
+                Collections.emptySortedSet(),
+                Collections.emptySortedSet(),
+                Collections.emptySortedSet(),
+                Collections.emptySortedSet()
+        );
+        Map<String, String> aliases = PluginUtils.computeAliases(result);
+        assertEquals(CollidingConverter.class.getName(), aliases.get("CollidingConverter"));
+        assertFalse(aliases.containsKey("Colliding"));
```

Comment:
Same thought RE comparing whole `Map` objects:

```suggestion
        Map<String, String> actualAliases = PluginUtils.computeAliases(result);

        Map<String, String> expectedAliases = new HashMap<>();
        expectedAliases.put("CollidingConverter", CollidingConverter.class.getName());
        assertEquals(expectedAliases, actualAliases);
```